tests/e2e/redis/parallel-ingest-redis-serving.py

```diff
@@ -0,0 +1,185 @@
+import os
+import uuid
+from datetime import datetime
+
+import pytest
+from google.protobuf.duration_pb2 import Duration
+
+from feast.client import Client
+from feast.data_source import DataSource, FileOptions, SourceType
+from feast.entity import Entity
+from feast.feature import Feature
+from feast.feature_table import FeatureTable
+from feast.value_type import ValueType
+
+DIR_PATH = os.path.dirname(os.path.realpath(__file__))
+PROJECT_NAME = "basic_" + uuid.uuid4().hex.upper()[0:6]
+
+
+@pytest.fixture(scope="module")
+def client(pytestconfig):
+    core_url = pytestconfig.getoption("core_url")
+    serving_url = pytestconfig.getoption("serving_url")
+
+    client = Client(core_url=core_url, serving_url=serving_url,)
+
+    client.set_project(PROJECT_NAME)
+
+    return client
+
+
+@pytest.mark.incremental
+class TestBasicIngestionRetrieval:
+    def setup_class(cls):
+        prefix = "basic_ingestion"
+        suffix = str(int(datetime.now().timestamp()))
+        cls.customer_ft_name = f"{prefix}_customer_{suffix}"
+        cls.driver_ft_name = f"{prefix}_driver_{suffix}"
+
+        cls.customer_entity = Entity(
+            name="customer_id",
+            description="Customer entity for rides",
+            value_type=ValueType.STRING,
+            labels={"team": "customer_service", "common_key": "common_val"},
+        )
+
+        cls.driver_entity = Entity(
+            name="driver_id",
+            description="Driver entity for car rides",
+            value_type=ValueType.STRING,
+            labels={"team": "matchmaking", "common_key": "common_val"},
+        )
+
+        cls.basic_ft_spec = FeatureTable.from_yaml(
+            f"{DIR_PATH}/specifications/dev_ft.yaml"
+        )
+
+    def test_discovery(self, client):
```

Comment:
Discovery as a name is overloaded. Can we be more specific on what we are testing?

Reply:
Renamed the tests.